-->> RDP Wrapper & Autoupdate (2024-04-29)

asmtron opened the pull request:
# RDP Wrapper & Autoupdate

### Automatic RDP Wrapper installer and updater

#### Info:
The autoupdater first use and check the official rdpwrap.ini. If a new termsrv.dll is not supported in the offical rdpwrap.ini, autoupdate uses the updated rdpwrap.ini files from the community.
Contributors: "sebaxakerhtc, asmtron, affinityv, DrDrrae, saurav-biswas"
Extra rdpwrap.ini sources can also be defined...
   
#### autoupdate.bat Options:
- -log        = redirect display output to the file autoupdate.log
- -taskadd    = add autorun of autoupdate.bat on startup in schedule task
- -taskremove = remove autorun of autoupdate.bat on startup in schedule task

#### [Download and Install](https://github.com/asmtron/rdpwrap/blob/master/binary-download.md)

#### Special thx to _binarymaster_ for creating this project and _sebaxakerhtc_ for all the rdpwrap.ini updates :)

###### *(Why this new pull-request ... removes the wrong edtrejo/rdpwrap connection ... last pull https://github.com/stascorp/rdpwrap/pull/859) *

asmtron committed:
Update README.md

asmtron committed:
Add files via upload

asmtron committed:
asmtron

asmtron committed:
asmtron

asmtron committed:
asmtron

asmtron committed:
asmtron

asmtron committed:
asmtron

asmtron committed:
Update README.md

asmtron committed:
Fix for Windows home editions

Windows home editions are missing the command "%windir%\system32\query.exe" to query the rdp-tcp session. The workaround on windows home editions is to check the windows registry value HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Terminal Server\fDenyTSConnections

asmtron committed:
Fix for Windows Home editions

asmtron commented:
@ ItachiSan 
the new autoupdate (12-09-2020) fixes the Windows Home "query.exe" missing problem

asmtron committed:
add additional new-line at the end

add additional new-line at the end to minimize the copy and past problems on the web browser

Adeshwar03 commented:
needed for 10.0.14393.3986

Clasen00 commented:
Thank you so much! You help a lot!

asmtron committed:
add "10.0.19041.662" support (x64)

asmtron committed:
fix error in 10.0.19041.662-SLInit

asmtron committed:
add support for 10.0.19041.662 (x64)

asmtron committed:
add 10.0.19041.662 (x86)

asmtron committed:
add support for 10.0.17763.1613 (x64)

asmtron committed:
Update binary-download.md

asmtron committed:
Update autoupdate__info.txt

asmtron committed:
Update autoupdate.bat

- changed the restart method so we can copy and not stream the new downloaded ini; hope this fixes problems with loading of the new ini file on some systems...
- add no-admin warning with pause,

asmtron committed:
RDP Wrapper Autoupdate

rctgamer3 commented:
While the auto-update script feature works, there are various things that should be addressed before calling the auto-update script stable. In its current form, take caution while using it.

- It deletes ALL existing firewall rules without notification (Great. I just lost my entire RDP whitelist in the firewall rule while updating)
- It doesn't make a backup of the existing RDP firewall rule(s)
- It assumes RDP is running on port 3389, the autoupdate script should be able to read out the port before setting things
- The autoupdate should have its own readme.md and not replace the first few lines of the README main project. Keep things separate.
- Probably more.

rctgamer3 commented:
It seems like these were existing issues with RDPWrap itself, not the update script. Still, weird functionality. Deleted my comment.

rajan-sounder commented:
Thank you @asmtron and the whole RDP Wrapper team who developed this awesome tool and continues to be maintained.
I spent quite time (around 8-10 hours) figuring out how to make it working once again after a major Windows update happened a couple of days ago which probably updated a new termsrv.dll.

Thanks to the below documentation link, it worked exactly as documented. 
Just strictly follow each instruction and worked like a charm 👍 :-)
https://github.com/stascorp/rdpwrap/blob/524a255e6ee80f11b0efd47f3729c74531c20484/binary-download.md

My Windows version on which above instructions worked was "Windows 10 Home x64, Version 2004 (OS Build 19041.685)"

Whaileee commented:
@rajan-sounder worked perfectly, issue solved. Also Windows 10 Home x64, Version 2004 (OS Build 19041.685)

Pythq committed:
Fixes typo in redirected error stream

autoupdate.bat created a file called "NULL".

audio99 commented:
how to disable rdpwrap logfile?
![IMG_20210110_022706_128](https://user-images.githubusercontent.com/9496025/104110173-84a1ba00-5307-11eb-8985-0ddc657f49ad.jpg)

i already leave blank the logfile save path but still present even i had delete it before and restart the pc

asmtron committed:
Merge pull request #3 from Pythq/master

Fixes typo in redirected error stream

asmtron committed:
Fixes typo in redirected error stream

asmtron commented:
@audio99
if you mean the logs of autoupdate.bat: remove the `-log` entry in the line `schtasks /create /f /sc ONSTART /tn "RDP Wrapper Autoupdate" /tr "cmd.exe /C \"%~dp0autoupdate.bat\" -log" /ru SYSTEM /delay 0000:10` and the run `autoupdate__enable_autorun_on_startup.bat` again.

asmtron committed:
add 10.0.18362.1316 (x64) and 10.0.19041.746 (x64)

asmtron committed:
add "Wait for network connection" :: important fix

Problem: on the new Win10 ver 10.0.19041.746 on startup the network is not ready for longer time; added the "wait for network connection" functionality to fix this issue

asmtron committed:
add "Wait for network connection" :: important fix 

Problem: on the new Win10 ver 10.0.19041.746 on startup the network is not ready for longer time; added the "wait for network connection" functionality to fix this issue

asmtron committed:
Update README.md

asmtron commented:
### _**autoupdate fix -> please update your autoupdate file (13-01-2021)**_ 

**add "Wait for network connection" :: important fix** 
Problem: on the new Win10 ver 10.0.19041.746 on startup the network is not ready for longer time; added the "wait for network connection" functionality to fix this issue

asmtron committed:
add 10.0.18362.1316, 10.0.19041.746, 10.0.17763.1697 (x64)

Cybernose commented:
I am very appreciative of asmtron, and the team continued support of this project and especially autoupdate.  I was having an issue this am and rebooted the computer, and all was well again.

asmtron committed:
add 10.0.18362.1316, 10.0.19041.746, 10.0.17763.1697, 10.0.14393.4169 (x64)

Adeshwar03 commented:
Heads up and warning.. I downloaded the autoupdate file and got RDP working again after the windows 10 update, within hours a ransomware attacked my system, tried to disable antivirus, changed the password of the default administrator account, installed a program called IOTunlocker or something like that. It was some Phobos ransomware as diagnosed by windows defender and the description of the ransomware says RDP is a source for this. I don't know if the 2 are connected or not, since I have been using the commits on this thread for over 2 yrs now and never faced any trouble. Circumstances overlapped and I thought it best to bring it to the notice of the forum.. Hope they are not connected. Any inputs on how to reduce such attacks and make it more secure are welcome..

systracer commented:
> Heads up and warning..

Check the source code to make sure this program is not relevant to the problem.

ZeljkoManjkas commented:
I found offsets for [10.0.18362.1316] x86.

Who can add this definition to rdpwrap.ini downloaded with autoupdate?
[rdpwrap.zip](https://github.com/stascorp/rdpwrap/files/5831081/rdpwrap.zip)


[10.0.18362.1316]
LocalOnlyPatch.x86=1
LocalOnlyOffset.x86=B67C6
LocalOnlyCode.x86=jmpshort
LocalOnlyPatch.x64=1
LocalOnlyOffset.x64=83075
LocalOnlyCode.x64=jmpshort
SingleUserPatch.x86=1
SingleUserOffset.x86=35735
SingleUserCode.x86=nop
SingleUserPatch.x64=1
SingleUserOffset.x64=0DC4C
SingleUserCode.x64=Zero
DefPolicyPatch.x86=1
DefPolicyOffset.x86=4D679
DefPolicyCode.x86=CDefPolicy_Query_eax_ecx
DefPolicyPatch.x64=1
DefPolicyOffset.x64=1FE65
DefPolicyCode.x64=CDefPolicy_Query_eax_rcx
SLInitHook.x86=1
SLInitOffset.x86=5C18A
SLInitFunc.x86=New_CSLQuery_Initialize
SLInitHook.x64=1
SLInitOffset.x64=22E9C
SLInitFunc.x64=New_CSLQuery_Initialize

[10.0.18362.1316-SLInit]
bInitialized.x86      =D377C
bServerSku.x86        =D3780
lMaxUserSessions.x86  =D3784
bAppServerAllowed.x86 =D378C
bRemoteConnAllowed.x86=D3790
bMultimonAllowed.x86  =D3794
ulMaxDebugSessions.x86=D3798
bFUSEnabled.x86       =D379C

bInitialized.x64      =F6A8C
bServerSku.x64        =F6A90
lMaxUserSessions.x64  =F6A94
bAppServerAllowed.x64 =F6A9C
bRemoteConnAllowed.x64=F6AA0
bMultimonAllowed.x64  =F6AA4
ulMaxDebugSessions.x64=F6AA8
bFUSEnabled.x64       =F6AAC

asmtron committed:
fix ping for IPv6 (check network connectivity)

asmtron committed:
fix ping for IPv6 (check network connectivity)

asmtron committed:
Update autoupdate__info.txt

asmtron committed:
Update binary-download.md

asmtron committed:
add 10.0.18362.1316(x86+x64); 10.0.19041.746(x64), 10.0.17763.1697(x64), 10.0.14393.4169(x64)

asmtron commented:
@brechtm
thx, added the fix (ping IPv6 address) on [autoupdate.bat](https://github.com/asmtron/rdpwrap/blob/master/bin/autoupdate.bat)
@ZeljkoManjkas
thx, offsets for x86 are correct, added to [rdpwrap.ini](https://github.com/asmtron/rdpwrap/blob/master/res/rdpwrap.ini)

ZeljkoManjkas commented:
Thanks

Today I was tested [10.0.19041.746] x86 

[10.0.19041.746]
LocalOnlyPatch.x86=1
LocalOnlyOffset.x86=B5979
LocalOnlyCode.x86=jmpshort
LocalOnlyPatch.x64=1
LocalOnlyOffset.x64=88F31
LocalOnlyCode.x64=jmpshort
SingleUserPatch.x86=1
SingleUserOffset.x86=3BC05
SingleUserCode.x86=nop
SingleUserPatch.x64=1
SingleUserOffset.x64=0CA4C
SingleUserCode.x64=Zero
DefPolicyPatch.x86=1
DefPolicyOffset.x86=3E779
DefPolicyCode.x86=CDefPolicy_Query_eax_ecx
DefPolicyPatch.x64=1
DefPolicyOffset.x64=18A15
DefPolicyCode.x64=CDefPolicy_Query_eax_rcx
SLInitHook.x86=1
SLInitOffset.x86=67B9A
SLInitFunc.x86=New_CSLQuery_Initialize
SLInitHook.x64=1
SLInitOffset.x64=1D5BC
SLInitFunc.x64=New_CSLQuery_Initialize

[10.0.19041.746-SLInit]
bInitialized.x86      =D0954
bServerSku.x86        =D0958
lMaxUserSessions.x86  =D095C
bAppServerAllowed.x86 =D0964
bRemoteConnAllowed.x86=D096C
bMultimonAllowed.x86  =D0970
ulMaxDebugSessions.x86=D0974
bFUSEnabled.x86       =D0978

bInitialized.x64      =106028
bServerSku.x64        =10602C
lMaxUserSessions.x64  =106030
bAppServerAllowed.x64 =106038
bRemoteConnAllowed.x64=106040
bMultimonAllowed.x64  =106044
ulMaxDebugSessions.x64=106048
bFUSEnabled.x64       =10604C

Also in ziped file 
[rdpwrap.zip](https://github.com/stascorp/rdpwrap/files/5835804/rdpwrap.zip)

r3d1r commented:
Hi asmtron.

What about change line 89 in _autoupdate.bat_ from 
   

> _pause_

to 
   

> _timeout 60_

in order to avoid an infinite pause when running as a scheduled task.

Regards.

some1o1 commented:
anyone can help .. win10 pro 32bit - version 19041.746

cannot concurrent users.. all check is green color but when connect its not allow concurrent user... 
did try auto update also not fix the issue ..

ZeljkoManjkas commented:
> anyone can help .. win10 pro 32bit - version 19041.746
>
> cannot concurrent users.. all check is green color but when connect its not allow concurrent user...
> did try auto update also not fix the issue ..

Fixing concurrent  users is above my knowledge.
Concurrent users doesn't work also on Win 10 pro 64bit version 19041.746 (some other found offsets values).

some1o1 commented:
> > anyone can help .. win10 pro 32bit - version 19041.746
> > cannot concurrent users.. all check is green color but when connect its not allow concurrent user...
> > did try auto update also not fix the issue ..
> 
> Fixing concurrent users is above my knowledge.
> Concurrent users doesn't work also on Win 10 pro 64bit version 19041.746 (some other found offsets values).

After i add the below code and restart pc ..its can work ready

[10.0.19041.746]
LocalOnlyPatch.x86  =1
LocalOnlyOffset.x86 =B5979
LocalOnlyCode.x86   =jmpshort
SingleUserPatch.x86 =1
SingleUserOffset.x86=3BC05
SingleUserCode.x86  =nop
DefPolicyPatch.x86  =1
DefPolicyOffset.x86 =3E779
DefPolicyCode.x86   =CDefPolicy_Query_eax_ecx
SLInitHook.x86      =1
SLInitOffset.x86    =67B9A
SLInitFunc.x86      =New_CSLQuery_Initialize

LocalOnlyPatch.x64  =1
LocalOnlyOffset.x64 =88F31
LocalOnlyCode.x64   =jmpshort
SingleUserPatch.x64 =1
SingleUserOffset.x64=0CB22
SingleUserCode.x64  =Zero
DefPolicyPatch.x64  =1
DefPolicyOffset.x64 =18A15
DefPolicyCode.x64   =CDefPolicy_Query_eax_rcx
SLInitHook.x64      =1
SLInitOffset.x64    =1D5BC
SLInitFunc.x64      =New_CSLQuery_Initialize

[10.0.19041.746-SLInit]
bInitialized.x86      =D0954
bServerSku.x86        =D0958
lMaxUserSessions.x86  =D095C
bAppServerAllowed.x86 =D0964
bRemoteConnAllowed.x86=D096C
bMultimonAllowed.x86  =D0970
ulMaxDebugSessions.x86=D0974
bFUSEnabled.x86       =D0978

bInitialized.x64      =106028
bServerSku.x64        =10602C
lMaxUserSessions.x64  =106030
bAppServerAllowed.x64 =106038
bRemoteConnAllowed.x64=106040
bMultimonAllowed.x64  =106044
ulMaxDebugSessions.x64=106048
bFUSEnabled.x64       =10604C

ZeljkoManjkas commented:
Hi asmtron.

Can you add [10.0.19041.746] x86 definition in rdpwrap.ini that can be downloaded with autoupdate?

This version was tested on two machines working two days.

asmtron committed:
add 10.0.18362.1316;10.0.19041.746,10.0.17763.1697,10.0.14393.4169

 add 10.0.18362.1316(x86+x64); 10.0.19041.746(x86+x64), 10.0.17763.1697(x86+x64), 10.0.14393.4169(x64)

asmtron commented:
add 10.0.19041.746 (x86) and 10.0.17763.1697 (x86) to [rdpwrap.ini](https://github.com/asmtron/rdpwrap/blob/master/res/rdpwrap.ini)

[Download & Install](https://github.com/asmtron/rdpwrap/blob/master/binary-download.md)

asmtron committed:
add 10.0.18362.1316;10.0.19041.746,10.0.17763.1697,10.0.14393.4169

 add 10.0.18362.1316(x86+x64); 10.0.19041.746(x86+x64), 10.0.17763.1697(x86+x64), 10.0.14393.4169(x64)

asmtron committed:
add 10.0.18362.1316;10.0.19041.746,10.0.17763.1697,10.0.14393.4169

add 10.0.18362.1316(x86+x64); 10.0.19041.746(x86+x64), 10.0.17763.1697(x86+x64), 10.0.14393.4169(x64)
...removed wrong entries in [main]

3wz commented:
中国用户建议开着vpn使用该批处理脚本，否则可能导致脚本运行无效
推荐另外一个地址：
https://gist.githubusercontent.com/javedsumaro/11fcabdf595241148ab962660ed470cc/raw/b5b1ff7a9cb72dea84b4dad3697680d0846aa62f/rdpwrap.ini

nagyimre1980 commented:
Hi asmtron.

Can you add [10.0.19041.789] x86 definition in rdpwrap.ini that can be downloaded with autoupdate?

asmtron committed:
add support for 10.0.19041.789

asmtron committed:
add support for 10.0.19041.789

ZeljkoManjkas commented:
I was tested [10.0.19041.789] x86 and works.

HYBRID-BEING committed:
Update binary-download.md

Fixed few typos.

asmtron committed:
add 10.0.10240.18818

asmtron committed:
add new ini source (contributor sebaxakerhtc)

add sebaxakerhtc as new rdpwrap.ini contributor (special thx to sebaxakerhtc)
changed "pause" to "timeout 60" (thx to r3d1r)
fixed typo

asmtron committed:
add new ini source (contributor sebaxakerhtc)

add sebaxakerhtc as new rdpwrap.ini contributor (special thx to sebaxakerhtc)
changed "pause" to "timeout 60" (thx to r3d1r)
fixed typo

asmtron committed:
add 10.0.10240.18818, 10.0.17134.1967 (x64)

asmtron committed:
fix typo

asmtron committed:
add new ini source (contributor sebaxakerhtc)

asmtron committed:
add new ini source (contributor sebaxakerhtc)

heylix committed:
fix link to Product Policy Editor, some formatting

LanofBlue commented:
> # RDP Wrapper & Autoupdate
> ### Automatic RDP Wrapper installer and updater // asmtron (2021-02-06)
> #### Info:
> The Autoupdater (autoupdate.bat) first use and check the official rdpwrap.ini.
> If a new termsrv.dll is not supported in the offical rdpwrap.ini,
> autoupdater first tries the asmtron rdpwrap.ini. The autoupdater will also use rdpwrap.ini files
> of other contributors. Extra rdpwrap.ini sources can also be defined...
> 
> #### autoupdate.bat Options:
> * -log        = redirect display output to the file autoupdate.log
> * -taskadd    = add autorun of autoupdate.bat on startup in schedule task
> * -taskremove = remove autorun of autoupdate.bat on startup in schedule task
> 
> #### [Download and Install](https://github.com/asmtron/rdpwrap/blob/master/binary-download.md)
> #### Special thx to binarymaster for creating this project :)
> ###### *(Why this new pull-request ... removes the wrong edtrejo/rdpwrap connection ... last pull #859) *



> # RDP Wrapper & Autoupdate
> ### Automatic RDP Wrapper installer and updater // asmtron (2021-02-06)
> #### Info:
> The Autoupdater (autoupdate.bat) first use and check the official rdpwrap.ini.
> If a new termsrv.dll is not supported in the offical rdpwrap.ini,
> autoupdater first tries the asmtron rdpwrap.ini. The autoupdater will also use rdpwrap.ini files
> of other contributors. Extra rdpwrap.ini sources can also be defined...
> 
> #### autoupdate.bat Options:
> * -log        = redirect display output to the file autoupdate.log
> * -taskadd    = add autorun of autoupdate.bat on startup in schedule task
> * -taskremove = remove autorun of autoupdate.bat on startup in schedule task
> 
> #### [Download and Install](https://github.com/asmtron/rdpwrap/blob/master/binary-download.md)
> #### Special thx to binarymaster for creating this project :)
> ###### *(Why this new pull-request ... removes the wrong edtrejo/rdpwrap connection ... last pull #859) *

Hi asmtron.

Can you add [10.0.19042.804] x86 definition in rdpwrap.ini？thx a lot

asmtron committed:
add support for termsrv.dll 10.0.14393.4169 (x86)

asmtron committed:
Update autoupdate__info.txt

asmtron committed:
improved -addtask command line parameter

add option AllowStartIfOnBatteries in the "startup schedule task" of "autoupdate.bat" ->fix: autoupdate will now also run on startup if the computer is on battery operation.
replaced batch comment command "REM" with "::"

asmtron committed:
improved -addtask command line parameter

add option AllowStartIfOnBatteries in the "startup schedule task" of "autoupdate.bat" ->fix: autoupdate will now also run on startup if the computer is on battery operation.
replaced batch comment command "REM" with "::"

asmtron committed:
Merge pull request #4 from HYBRID-BEING/patch-1

thx

asmtron committed:
Merge pull request #5 from heylix/patch-1

fix link to Product Policy Editor, some formatting

serj-lb commented:
Не работает 10.0.17763.1369

asmtron committed:
add support for termsrv.dll 10.0.19041.964 x64

asmtron committed:
add support termsrv.dll v10.0.19041.964 (x86 and x64)

asmtron committed:
add termsrv.dll v10.0.19041.964 (x86+x64), 10.0.18362.1533 (x64)

ZeljkoManjkas commented:
New debuged and tested definition for [10.0.18362.1533] x86

[rdpwrap.zip](https://github.com/stascorp/rdpwrap/files/6486530/rdpwrap.zip)
.

asmtron committed:
add 10.0.19041.1023 (x64), 10.0.18362.1533 (x86)

thx to ZeljkoManjkas for 10.0.18362.1533 (x86)
thx to sebaxakerhtc for 10.0.19041.1023 (x64) - note: the SingleUserOffset.x64=0CA4C is different

SergiuEnachi commented:
please add support for build **10.0.17763.1971** OS Server 2019 v1809

ZeljkoManjkas commented:
I have new debuged and tested definition for [10.0.19041.1023] x86
[rdpwrap.zip](https://github.com/stascorp/rdpwrap/files/6584756/rdpwrap.zip)

ZeljkoManjkas commented:
Can you update ini file with [10.0.19041.1023] x86 definition?

some1o1 commented:
Hi ,

seems all ok but we still cannot use ... win10 32bit ..
when connect it show error it say another use login ..so the concurrent login cannot work with new win10 version 



From: asmtron 
Sent: Sunday, January 24, 2021 5:51 PM
To: stascorp/rdpwrap 
Cc: some1o1 ; Comment 
Subject: Re: [stascorp/rdpwrap] -->> RDP Wrapper & Autoupdate (18-01-2021) (#1160)

add 10.0.19041.746 (x86) and 10.0.17763.1697 (x86) to rdpwrap.ini

Download & Install

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.

ZeljkoManjkas commented:
I found values for 10.0.19041.1081 x86 that works on Windows 10 2004 and 20h2.
Can you update this section-s to autoupdate?

[rdpwrap.zip](https://github.com/stascorp/rdpwrap/files/6794788/rdpwrap.zip)

asmtron committed:
add support for 10.0.19041.1081 (x86 and x64)

thx to sebaxakerhtc and Zeljko Manjkas

everdominguez commented:
> I found values for 10.0.19041.1081 x86 that works on Windows 10 2004 and 20h2.
> Can you update this section-s to autoupdate?
> 
> [rdpwrap.zip](https://github.com/stascorp/rdpwrap/files/6794788/rdpwrap.zip)

¡Thanks! This works for multi session in 1081. Just need download the new rdwrap.ini and remplace it. After that, reboot and thats it.

jonathan19091 commented:
ya salió la 10.0.19041.1202 y la cual de nuevo dejo fuera a rdp, veremos como nos va.

rohinmehra commented:
Please provide support for Windows 11 Pro Insider Preview OS Build 22449.1000. Many thanks!

naider42 commented:
Please provide support for Windows 10 10.19041.1202

jonathan19091 commented:
I'm already on the newest version of windows 11 pro,

Outlook para Android<https://aka.ms/ghei36>
________________________________
From: naider42 ***@***.***>
Sent: Wednesday, September 15, 2021 10:12:19 PM
To: stascorp/rdpwrap ***@***.***>
Cc: jonathan19091 ***@***.***>; Comment ***@***.***>
Subject: Re: [stascorp/rdpwrap] -->> RDP Wrapper & Autoupdate (2021-04-26) (#1160)


Please provide support for Windows 10 10.19041.1202

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/stascorp/rdpwrap/pull/1160#issuecomment-920541483>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALAQDX4RSLZ6HRGEFPT7SE3UCFOBHANCNFSM4Q4F6L7A>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

jonathan19091 commented:
Windows 11 pro RDP works well
________________________________
From: naider42 ***@***.***>
Sent: Wednesday, September 15, 2021 10:12:19 PM
To: stascorp/rdpwrap ***@***.***>
Cc: jonathan19091 ***@***.***>; Comment ***@***.***>
Subject: Re: [stascorp/rdpwrap] -->> RDP Wrapper & Autoupdate (2021-04-26) (#1160)


Please provide support for Windows 10 10.19041.1202

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/stascorp/rdpwrap/pull/1160#issuecomment-920541483>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALAQDX4RSLZ6HRGEFPT7SE3UCFOBHANCNFSM4Q4F6L7A>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

naider42 commented:
But I need windows 10 Pro I can't format the version I need and support for Windows 10 10.19041.1202 can help me?

rohinmehra commented:
Not working on Windows 11 Home Insider build 22458

jonathan19091 commented:
Only in pro version

Outlook para Android<https://aka.ms/ghei36>
________________________________
From: rohinmehra ***@***.***>
Sent: Thursday, September 16, 2021 2:31:08 PM
To: stascorp/rdpwrap ***@***.***>
Cc: jonathan19091 ***@***.***>; Comment ***@***.***>
Subject: Re: [stascorp/rdpwrap] -->> RDP Wrapper & Autoupdate (2021-04-26) (#1160)


Not working on Windows 11 Home Insider build 22458

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/stascorp/rdpwrap/pull/1160#issuecomment-921186900>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALAQDXZ2GZLLIZZXTWFBPM3UCJAXZANCNFSM4Q4F6L7A>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

jcalem14 commented:
Hello.

Windows version 21H2 (OS Build 2000.194)

![Screenshot 2021-10-20 115043](https://user-images.githubusercontent.com/92876362/138127423-7515d47a-8749-4f63-b67f-01c472002b57.jpg)
![Screenshot 2021-10-20 115138](https://user-images.githubusercontent.com/92876362/138127533-1dfb57c1-5692-475c-a5d6-9d584e393e6f.jpg)

csacksteder commented:
> Please provide support for Windows 10 10.19041.1202

In https://github.com/stascorp/rdpwrap/issues/1509 someone has posted a suggested entry, which would be nice to have in the copy autoupdate.cmd downloads (e.g., on remote systems I'd rather not reboot).

Quote:
10.0.19041.1081 and 10.0.19041.1202 use the same settings. Append these to your rdpwrapper.ini file.

[10.0.19041.1081]
LocalOnlyPatch.x64=1
LocalOnlyOffset.x64=89D81
LocalOnlyCode.x64=jmpshort
SingleUserPatch.x64=1
SingleUserOffset.x64=0CB26
SingleUserCode.x64=Zero
DefPolicyPatch.x64=1
DefPolicyOffset.x64=19105
DefPolicyCode.x64=CDefPolicy_Query_eax_rcx
SLInitHook.x64=1
SLInitOffset.x64=1E98C
SLInitFunc.x64=New_CSLQuery_Initialize

[10.0.19041.1081-SLInit]
bInitialized.x64 =107108
bServerSku.x64 =10710C
lMaxUserSessions.x64 =107110
bAppServerAllowed.x64 =107118
bRemoteConnAllowed.x64=107120
bMultimonAllowed.x64 =107124
ulMaxDebugSessions.x64=107128
bFUSEnabled.x64 =10712C

[10.0.19041.1202]
LocalOnlyPatch.x64=1
LocalOnlyOffset.x64=89D81
LocalOnlyCode.x64=jmpshort
SingleUserPatch.x64=1
SingleUserOffset.x64=0CB26
SingleUserCode.x64=Zero
DefPolicyPatch.x64=1
DefPolicyOffset.x64=19105
DefPolicyCode.x64=CDefPolicy_Query_eax_rcx
SLInitHook.x64=1
SLInitOffset.x64=1E98C
SLInitFunc.x64=New_CSLQuery_Initialize

[10.0.19041.1202-SLInit]
bInitialized.x64 =107108
bServerSku.x64 =10710C
lMaxUserSessions.x64 =107110
bAppServerAllowed.x64 =107118
bRemoteConnAllowed.x64=107120
bMultimonAllowed.x64 =107124
ulMaxDebugSessions.x64=107128
bFUSEnabled.x64 =10712C

csacksteder commented:
Oh, I see rdpwrap.ini has been updated by sebaxakerhtc.  Thanks!!!

Oddly, on most systems, I had to use an edited version of autoupdate.bat that set https://raw.githubusercontent.com/sebaxakerhtc/rdpwrap.ini/master/rdpwrap.ini to rdpwrap_ini_update_github_1 rather than rdpwrap_ini_update_github_2.  Otherwise I was getting the old version rdpwrap.ini that did not have an entry matching the .dll.  I don't see why, as the log says it can't find an entry for 10.0.19041.12002 and I think it should move through each URL.

LookingforRDPhelp commented:
> # RDP Wrapper & Autoupdate
> ### Automatic RDP Wrapper installer and updater
> #### Info:
> The Autoupdater (autoupdate.bat) first use and check the official rdpwrap.ini. If a new termsrv.dll is not supported in the offical rdpwrap.ini the autoupdater will use rdpwrap.ini files of other contributors: asmtron, sebaxakerhtc, affinityv... Extra rdpwrap.ini sources can also be defined...
> 
> #### autoupdate.bat Options:
> * -log        = redirect display output to the file autoupdate.log
> * -taskadd    = add autorun of autoupdate.bat on startup in schedule task
> * -taskremove = remove autorun of autoupdate.bat on startup in schedule task
> 
> #### [Download and Install](https://github.com/asmtron/rdpwrap/blob/master/binary-download.md)
> #### Special thx to binarymaster for creating this project :)
> ###### *(Why this new pull-request ... removes the wrong edtrejo/rdpwrap connection ... last pull #859) *

i downloaded this but everytime i try to load it up it instantly exits out of it and i have the same problem with rdpwlnst

csacksteder commented:
Today I'm not seeing entries for 19041.1348 . . . looked at various locations and found what looks to be the most recent version in https://raw.githubusercontent.com/affinityv/INI-RDPWRAP/master/rdpwrap.ini.  I put that in autoupdate.bat as "set rdpwrap_ini_update_github_1=..." and now that works.

Autoupdate.bat used to be flawless. I don't understand how it all works and what happened.

black-zzy commented:
![image](https://user-images.githubusercontent.com/83347879/143334584-cc09285a-7aaa-4949-9f72-f189670f107f.png)
I don't know how to solve this problem,I'v tried many times,but I can't

black-zzy commented:
> ![image](https://user-images.githubusercontent.com/83347879/143334584-cc09285a-7aaa-4949-9f72-f189670f107f.png) I don't know how to solve this problem,I'v tried many times,but I can't

![image](https://user-images.githubusercontent.com/83347879/143338734-9b487ab8-b6a8-4349-8594-dd199a256fdd.png)

black-zzy commented:
> > ![image](https://user-images.githubusercontent.com/83347879/143334584-cc09285a-7aaa-4949-9f72-f189670f107f.png) I don't know how to solve this problem,I'v tried many times,but I can't
> 
> ![image](https://user-images.githubusercontent.com/83347879/143338734-9b487ab8-b6a8-4349-8594-dd199a256fdd.png)

![image](https://user-images.githubusercontent.com/83347879/143342023-cc06ac2e-d3b8-45da-abc0-a2aa45f1512b.png)

asmtron committed:
improved autoupdate.bat

add check if rdpwrap.dll still exists **
add force uninstall of rdpwrap **

**required when parts of rdpwrap are falsely deleted/changed by third party tools, antivirus or other removal tools

asmtron committed:
improved autoupdate.bat

add check if rdpwrap.dll still exists **
add force uninstall of rdpwrap **

**required when parts of rdpwrap are falsely deleted/changed by third party tools, antivirus or other removal tools

asmtron committed:
Update rdpwrap.ini

add 10.0.19041.1200 to 10.0.19041.1387 for win10
special thanks to sebaxakerhtc for all the updates and all new ini entries!!!

asmtron commented:
**first of all I want to thank sebaxakerhtcfor and affinityv for all the rdpwrap.ini entry updates!!!**

The new improved autoupdate.bat (from 2021-12-16) additional check if rdpwrap.dll still exists and force the uninstall of rdpwrap. This is required when parts of rdpwrap are falsely deleted/changed by third party tools, antivirus or other removal tools. 

Hopefully this update should solve the {Listener state: Not listening but [fully supported]} **issues** with {Wrapper state: Installed} and {Service state: Running} correctly...

alryaz committed:
fix: quoted rdpwrap_force_uninstall

sumanthmattapali commented:
Thanks for wonderful work to the community. Are you guys planning to merge this pull request and create a new version rdpwrapper?

asmtron committed:
Merge pull request #8 from alryaz/patch-1

Quoted rdpwrap_force_uninstall (fixes installation)

asmtron committed:
fix rdpwrap_force_uninstall

asmtron committed:
fix rdpwrap_force_uninstall

Neustradamus commented:
Any news about it?

hahacandy commented:
thank you!!

tekert commented:
I lost access to remote machine while running this, on the first run it just closed the listen port. Had to travel to this machine because i could not figure out why from remote console, running the `RDPConf.exe` GUI showed up why.
It also deleted the Authetication Mode used, why override the registry?
Also getting code 5 errors, the `schtask` should be created with `/RL HIGHEST`

Attached log:
This was run from a machine that already had rdpwrap installed but with and outdated ini, (script was run for the first time), afterwards a remote reboot was done.
[autoupdate.log](https://github.com/stascorp/rdpwrap/files/12207785/autoupdate.log)

asmtron committed:
autoupdate 1.1 with version check

asmtron committed:
Delete res/autoupdate.bat

asmtron committed:
Delete res/autoupdate.ver

asmtron committed:
autoupdate 1.1 with new version check

asmtron committed:
Update autoupdate__info.txt

asmtron committed:
autoupdate 1.1 with new version check

asmtron committed:
Update README.md

asmtron committed:
autoupdate 1.1 with new version check

autoupdate 1.1 with new version check

asmtron committed:
autoupdate 1.1 with new version check

autoupdate 1.1 with new version check

asmtron committed:
old autoupdate (from 2022-01-01)

old autoupdate (from 2022-01-01)

asmtron committed:
new autoupdate 1.1 with version check 

new autoupdate 1.1 with version check

asmtron committed:
new autoupdate 1.1 with version check (2022-10-18)

new autoupdate 1.1 with version check (2022-10-18)

asmtron committed:
new autoupdate 1.1 with version check (2022-10-18)

new autoupdate 1.1 with version check (2022-10-18)

asmtron committed:
Update binary-download.md

asmtron committed:
Update README.md

asmtron committed:
autoupdate 1.1 with new version check

autoupdate 1.1 with new version check

asmtron committed:
Update autoupdate__info.txt

jonathan19091 commented:
10.0.22621.2706
[Uploading termsrv.zip…]()

asmtron committed:
Update rdpwrap.ini

add 10.0.22621.3235 x64

asmtron committed:
add 10.0.22621.3374

add 10.0.22621.3374

asmtron committed:
Update autoupdate.bat to v.1.2

change first ini source to asmtron

asmtron committed:
Update autoupdate.ver to 1.2

asmtron commented:
add termsrv.dll 10.0.22621.3374 support

asmtron committed:
Add files via upload

asmtron committed:
Delete autoupdate_v1.2.zip

asmtron committed:
autoupdate 1.2 with new version check

autoupdate 1.2 with new version check

asmtron committed:
Update binary-download.md

asmtron committed:
Update README.md

asmtron committed:
Update rdpwrap.ini

add 10.0.22621.3447

asmtron committed:
Update rdpwrap.ini

add 10.0.22621.3447

xXWarMachineRoXx commented:
Hi, is this repo still being maintained?

mrkacg1 commented:
Hello. any manual to update rdpwrap.ini to new termsrv.dll builds?

xXWarMachineRoXx commented:
> Hello. any manual to update rdpwrap.ini to new termsrv.dll builds?

Its really easy. Edit with notepad

mrkacg1 commented:
I change the question. How should the rdpwrap.ini file be updated with the new termsrv.dll updates? What parameters should be taken into account for the updates?

asmtron committed:
Update autoupdate.bat

add contributor "apolope" with new up-to-date rdpwrap.ini entries

asmtron committed:
Update autoupdate.bat

asmtron committed:
Update autoupdate.ver to 1.3

Update autoupdate.ver to 1.2

Rincewindwiz commented:
The whole concurrent RSP work has been simply amazing and hugely helpful. Many thanks to all concerned.
I am not clear if this is meant to allow more than one user logged in under the same username?
The Host machine is Win 7 Ultimate SP1 Build 7601
I was logged into the host machines with a particular account
I then logged in using RDP to the same account from a different machine. It all seemed to work fine but I keep getting an error message 'Unexpected error while waiting for connections from a second instance"
So, please
1. Is it allowed to have multiple logins to the same windows account and if so, why this error message
2. Or if it is not allowed, wonderful that it works but can I expect it to keep working?

Jazzman1994 commented:
hello, plz link me working  RDPwrapper version full pack... i have 2019 server windos. its ASAP

Rincewindwiz commented:
I (think I) have what you want but email identifiers it as a virus and wont
let me send it to you


Suggestions?

asmtron committed:
Update rdpwrap.ini

asmtron committed:
Update rdpwrap.ini, add autogen support [*.autogen]

RDPWrap-with-Autoupdate-v1.4 supports supports autogeneration of rdpwrap.ini entries.

asmtron committed:
Update autoupdate.bat, add autogen support [*.autogen]

asmtron committed:
Update autoupdate.ver to v1.4

asmtron committed:
add pdblister v0.0.4 (Symbol Checker tool)

RDPWrap-with-Autoupdate-v1.4 use the Symbol Checker tool pdblister v0.0.4 for autogeneration of rdpwrap.ini entries

asmtron committed:
add cvdump

RDPWrap-with-Autoupdate-v1.4 use the "Microsoft Debugging Information Dumper Version 14.00.23611" for autogen support [*.autogen]
(auto-generation of rdpwrap.ini entries)

asmtron committed:
Update autoupdate__info.txt

asmtron committed:
add cvdump (Microsoft Debugging Information Dumper v.14.00.23611)

Microsoft Debugging Information Dumper Version 14.00.23611

RDPWrap-with-Autoupdate-v1.4 use the "Microsoft Debugging Information Dumper Version 14.00.23611" for autogen support [*.autogen]
(auto-generation of rdpwrap.ini entries)

asmtron committed:
Update binary-download.md

asmtron committed:
Update README.md

asmtron committed:
Update autoupdate.bat, add autogen support [*.autogen]

fix tab remove end lines ^M

asmtron committed:
Update autoupdate.bat, add autogen support [*.autogen]

fix tabs, removed end lines ^M

asmtron committed:
Update autoupdate.bat, add autogen support [*.autogen]

fix tab and end lines

asmtron committed:
Update autoupdate.batUpdate autoupdate.bat, add autogen support [*.autogen]

add '"-update"' bugfix to support update from older older autoupdate v1.3 ...

asmtron committed:
Update autoupdate.bat, add autogen support [*.autogen]

add '"-update"' bugfix to support update from older autoupdate versions (v1.3 ...)
